cxx-qt-lib/src/qcolor.rs

```diff
@@ -60,153 +60,175 @@ pub struct QColor {
+    }
+
+    pub fn blue(&self) -> i32 {
+        // Safety: TODO
+        unsafe { qcolor_get_blue(&self.inner) }
+    }
+
+    pub fn green(&self) -> i32 {
+        // Safety: TODO
+        unsafe { qcolor_get_green(&self.inner) }
     }
-}
 
-impl Drop for StackQColor {
-    fn drop(&mut self) {
-        // # Safety
-        //
-        // This simply calls ~QColor on self.space which is safe as long
-        // as self.space contains a valid QColor. Using the [`let_qcolor!`]
-        // macro guarantees that this will be the case.
+    pub fn red(&self) -> i32 {
+        // Safety: TODO
+        unsafe { qcolor_get_red(&self.inner) }
+    }
+
+    pub fn set_alpha(&mut self, alpha: i32) {
+        // Safety: TODO
         unsafe {
-            let this = &mut *self.space.as_mut_ptr().cast::<MaybeUninit<QColor>>();
-            qcolor_drop(this);
+            qcolor_set_alpha(self.inner.pin_mut(), alpha);
         }
     }
-}
 
-// Safety:
-//
-// The code in this file ensures that QColor can only ever be allocated
-// on the stack in pinned form which avoids the pitfalls of trying to
-// move this type that has a non-trivial move constructor.
-unsafe impl ExternType for QColor {
-    type Id = type_id!("QColor");
-    type Kind = cxx::kind::Opaque;
-}
+    pub fn set_blue(&mut self, blue: i32) {
+        // Safety: TODO
+        unsafe {
+            qcolor_set_blue(self.inner.pin_mut(), blue);
+        }
+    }
+
+    pub fn set_green(&mut self, green: i32) {
+        // Safety: TODO
+        unsafe {
+            qcolor_set_green(self.inner.pin_mut(), green);
+        }
+    }
+
+    pub fn set_red(&mut self, red: i32) {
+        // Safety: TODO
+        unsafe {
+            qcolor_set_red(self.inner.pin_mut(), red);
+        }
+    }
 
-pub enum Color {
-    ARGB {
-        alpha: u16,
-        red: u16,
-        green: u16,
-        blue: u16,
-    },
+    /// Retrieve the UniquePtr to the Qt QColor of this Rust Color
+    /// so that this object can be passed back to C++.
+    pub fn to_unique_ptr(self) -> cxx::UniquePtr<QColor> {
+        self.inner
+    }
 }
 
-impl From<&QColor> for Option<Color> {
+impl From<&QColor> for Color {
     fn from(qcolor: &QColor) -> Self {
         qcolor.to_rust()
     }
 }
+
+extern "C" {
+    #[link_name = "cxxqt1$unique_ptr$qcolor$null"]
+    fn unique_ptr_qcolor_null(this: *mut MaybeUninit<*mut c_void>);
+    #[link_name = "cxxqt1$unique_ptr$qcolor$raw"]
+    fn unique_ptr_qcolor_raw(this: *mut MaybeUninit<*mut c_void>, raw: *mut QColor);
+    #[link_name = "cxxqt1$unique_ptr$qcolor$get"]
+    fn unique_ptr_qcolor_get(this: *const MaybeUninit<*mut c_void>) -> *const QColor;
+    #[link_name = "cxxqt1$unique_ptr$qcolor$release"]
+    fn unique_ptr_qcolor_release(this: *mut MaybeUninit<*mut c_void>) -> *mut QColor;
+    #[link_name = "cxxqt1$unique_ptr$qcolor$drop"]
+    fn unique_ptr_qcolor_drop(this: *mut MaybeUninit<*mut c_void>);
+}
+
+// Safety: TODO
+unsafe impl UniquePtrTarget for QColor {
```

Comment:
This impl, as well as the extern "C" block should probably be placed right after the QColor definition, not the Color definition. It might be very easy to miss that one "Q" and think this is actually implemented on "Color".

So either move this up, or move the QColor impl down.

Reply:
Done